src/domain/accounts/services/src/services/device_code_service_impl.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use chrono::Duration;
+use init_on_startup::{InitOnStartup, InitOnStartupMeta};
+use internal_error::{InternalError, ResultIntoInternal};
+use kamu_accounts::{
+    CleanupExpiredDeviceCodesError,
+    CreateDeviceCodeError,
+    DeviceClientId,
+    DeviceCode,
+    DeviceCodeRepository,
+    DeviceCodeService,
+    DeviceToken,
+    DeviceTokenCreated,
+    DeviceTokenParamsPart,
+    FindDeviceTokenByDeviceCodeError,
+    UpdateDeviceCodeWithTokenParamsPartError,
+    JOB_KAMU_ACCOUNTS_DEVICE_CODE_SERVICE,
+    JOB_KAMU_ACCOUNTS_PREDEFINED_ACCOUNTS_REGISTRATOR,
+};
+use kamu_datasets::JOB_KAMU_DATASETS_DATASET_ENTRY_INDEXER;
+use time_source::SystemTimeSource;
+use uuid::Uuid;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const DEVICE_CODE_EXPIRES_IN: Duration = Duration::minutes(5);
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[dill::component]
+#[dill::interface(dyn DeviceCodeService)]
+#[dill::interface(dyn InitOnStartup)]
+#[dill::meta(InitOnStartupMeta {
+    job_name: JOB_KAMU_ACCOUNTS_DEVICE_CODE_SERVICE,
+    depends_on: &[
+        JOB_KAMU_ACCOUNTS_PREDEFINED_ACCOUNTS_REGISTRATOR,
+        JOB_KAMU_DATASETS_DATASET_ENTRY_INDEXER
```

Comment:
Why does it need any dataset entries at all?
I would rather avoid any dependencies of accounts domain from datasets domain, as the other direction of dependency exists already.